### PR TITLE
MolGANMultiConvolutionLayer Porting

### PR DESCRIPTION
**Pull request by shreyasvinaya**
## Description
Porting of MolGANMultiConvolutionLayer from TF to Torch
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -3103,7 +3103,7 @@ def __init__(self,
                  dropout_rate: float = 0.0,
                  edges: int = 5,
                  name: str = "",
-                 **kwargs):
+                 prev_shape: int = 0):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:3106]: Can you add new argument to the docs?

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -3132,12 +3132,27 @@ def __init__(self,
         self.name: str = name
         self.nodes: int = nodes
 
-        self.dense1: nn.ModuleList = nn.ModuleList(
-            [nn.Linear(nodes, self.units) for _ in range(edges - 1)])
+        # Case when >2 inputs are passed
+        if "prev_shape":
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:3136]: This is a type. I think you mean 
```
if prev_shape:
```
With a string, this branch will always activate

File: deepchem/models/torch_models/tests/test_molgan_layers.py [hunk 3]
```diff
@@ -145,3 +145,81 @@ def test_aggregation_layer_values():
...
+
+    torch.manual_seed(21)  # Setting seed for reproducibility
+    layer_multi_conv = MolGANMultiConvolutionLayer(units=units,
+                                                   nodes=nodes,
+                                                   edges=edges,
+                                                   name='layer1')
+    tf_weights = np.load(
+        'deepchem/models/tests/assets/molgan_multi_conv_layer_weights.npy',
+        allow_pickle=True).item()
+    with torch.no_grad():
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_molgan_layers.py:189]: Can you addd comments explaining the loops here? This is looking fairly complicated and I'm having trouble following

**Review by rbharath — COMMENTED** [review]
@shreyasvinaya I have a few comments and requests for clarification below

File: deepchem/models/torch_models/layers.py (hunk 1 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py:3106]: done

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py:3136]: yup, was a typo, fixed it

File: deepchem/models/torch_models/tests/test_molgan_layers.py (hunk 3 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/tests/test_molgan_layers.py:189]: done

Comment (rbharath) [reply deepchem/models/torch_models/tests/test_molgan_layers.py:189]: @shreyasvinaya Please hold off on resolving comments even if you have fixed the issue. I use comments to track needed changes and verify all needed work has been done

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**